src/persist-client/src/operators/shard_source.rs

```diff
@@ -380,7 +380,7 @@ where
             while inflight_bytes >= max_inflight_bytes {
                 // We can never get here when flow control is disabled, as we are not tracking
                 // in-flight bytes in this case.
-                assert_eq!(flow_control_bytes, None);
+                assert!(flow_control_bytes.is_some());
```

Comment:
I messed up the initial check... We never noticed because we didn't have flow control enabled before.